### PR TITLE
MWPW-146675 Check blog links

### PR DESCRIPTION
**Pull request by Brandon32**
* Report to find link shuffling throughout blog

Resolves: [MWPW-146675](https://jira.corp.adobe.com/browse/MWPW-146675)

### CODE REVIEW

File: link-check/caasLinks.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const list = await loadListData(LIST);
+  const { length } = list;
+
+  for (const [i, entry] of list.entries()) {
+    const percentage = Math.round(((i + 1) / length) * 10000) / 100;
+    console.log(`Processing entry ${i + 1} of ${length} (${percentage}%) ${entry}`);
+    const miloURL = MILO_BLOG + entry;
+    const franklinURL = FRANKLIN_BLOG + entry;
+    try {
+      const result = await compare(miloURL, franklinURL);
```
Comment (meganthecoder) [inline link-check/caasLinks.js:28]: It looks like compareLink doesn't compare the hashes. Should we include that? A lot of the old pages have the Table of Contents-style bullets with anchor links https://business.adobe.com/fr/blog/basics/8-wastes-of-lean

**Review by meganthecoder — COMMENTED** [review]
A couple comments. Also, I know tests aren't a priority right now but are we going to add them at some point?

File: link-check/caasLinks.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { compare } from './linkCompare.js';
+import { ExcelReporter, loadListData } from '../bulk-update/index.js';
+
+const { pathname } = new URL('.', import.meta.url);
```
[the remark this answers is not in the document]
Comment (Brandon32) [reply link-check/caasLinks.js]: `pathname` is not used for the report, it is used for getting the local path where the script is running so we can save the xlsx file.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import.meta#resolving_a_file_relative_to_the_current_one

File: link-check/caasLinks.js (hunk 1 above)
Comment (Brandon32) [reply link-check/caasLinks.js]: You can now pass a custom function into compare that returns true/false so any logic can be added to the checker.

File: link-check/linkCompare.js [hunk 3]
```diff
@@ -2,18 +2,25 @@ import fs from 'fs';
 import { fetch } from '@adobe/fetch';
 import { docx2md } from '@adobe/helix-docx2md';
 
+export function comparison(url1, url2) {
+  return (url1.host === url2.host) && (url1.pathname === url2.pathname);
```
Comment (JasonHowellSlavin) [inline link-check/linkCompare.js:6]: no change needed, but I wonder if we even need to compare hosts. I feel like you ran into this where `b.a.com` !== `main--bacom-blog--adobecom.hlx.page/live` and lord knows if even a full url was used.

**Review by JasonHowellSlavin — APPROVED** [review]

File: link-check/linkCompare.js (hunk 3 above)
Comment (Brandon32) [reply link-check/linkCompare.js:6]: This is an issue with some links and I started to address it but I'm moving that work to a separate ticket as we have already handed off the report.

**Pull request merged**